### PR TITLE
Fix exit manager exit change bug

### PR DESCRIPTION
**Pull request by ptulugu**
When an exit is manually changed, the last exit also changes and doesnt trigger teh linux setup exit tunnel function. Storing this in a lazy static instead of derviving each tick should account for manual changes to exits

### CODE REVIEW

File: rita_client/src/rita_loop/mod.rs [hunk 1]
```diff
@@ -138,10 +131,63 @@ pub fn start_rita_loop() {
...
+                        start.elapsed().subsec_millis()
+                    );
+                });
+
+                // sleep until it has been FAST_LOOP_SPEED seconds from start, whenever that may be
+                // if it has been more than FAST_LOOP_SPEED seconds from start, go right ahead
+                info!("Exit Manager loop elapsed in = {:?}", start.elapsed());
+                if start.elapsed() < EXIT_LOOP_SPEED {
+                    info!(
+                        "Peer listener sleeping for {:?}",
```
Comment (jkilpatr) [inline rita_client/src/rita_loop/mod.rs:163]: this isn't peer listener?

**Review by jkilpatr — CHANGES_REQUESTED** [review]
This looks great! One comment on log messages.

File: rita_client/src/rita_loop/mod.rs (hunk 1 above)
Comment (ptulugu) [reply rita_client/src/rita_loop/mod.rs]: fixed

File: rita_client/src/exit_manager/mod.rs [hunk 2]
```diff
@@ -72,78 +76,121 @@ pub struct ExitBlacklist {
...
 
-pub fn set_em_nat(val: bool) {
-    EXIT_MANAGER.write().unwrap().nat_setup = val;
+pub fn set_em_nat(val: bool, em_state: &mut ExitManager) {
+    em_state.nat_setup = val;
 }
 
-pub fn get_em_nat() -> bool {
-    EXIT_MANAGER.read().unwrap().nat_setup
+pub fn get_em_nat(em_state: &mut ExitManager) -> bool {
```
Comment (jkilpatr) [inline rita_client/src/exit_manager/mod.rs:183]: this is a mutable reference when it doesn't need to be, also I'm not sure the function needs to exist?

File: rita_client/src/exit_manager/mod.rs [hunk 3]
```diff
@@ -72,78 +76,121 @@ pub struct ExitBlacklist {
...
+    SELECTED_EXIT_LIST
         .write()
         .unwrap()
         .selected_exit_list
         .insert(exit, exit_info);
 }
 
-pub fn set_em_nat(val: bool) {
-    EXIT_MANAGER.write().unwrap().nat_setup = val;
+pub fn set_em_nat(val: bool, em_state: &mut ExitManager) {
```
Comment (jkilpatr) [inline rita_client/src/exit_manager/mod.rs:179]: I don't see what this function helps with?

File: rita_client/src/exit_manager/mod.rs [hunk 4]
```diff
@@ -72,78 +76,121 @@ pub struct ExitBlacklist {
...
+        .unwrap()
+        .selected_exit_list
+        .get(&exit)
+    {
         Some(a) => a.tracking_exit,
         None => None,
     }
 }
 
 pub fn get_selected_exit_degradation(exit: String) -> Option<u16> {
```
Comment (jkilpatr) [inline rita_client/src/exit_manager/mod.rs:159]: why not combine the above three functions and just return the whole exit struct?

File: rita_client/src/exit_manager/mod.rs [hunk 5]
```diff
@@ -72,78 +76,121 @@ pub struct ExitBlacklist {
...
+#[derive(Default, Clone)]
+pub struct ExitManager {
+    pub nat_setup: bool,
     /// Every tick we query an exit endpoint to get a list of exits in that cluster. We use this list for exit switching
     pub exit_list: ExitList,
+    /// Store last exit here, when we see an exit change, we reset wg tunnels
+    pub last_exit: Option<IpAddr>,
+}
+
+pub fn set_last_exit(e: Option<IpAddr>, em_state: &mut ExitManager) {
```
Comment (jkilpatr) [inline rita_client/src/exit_manager/mod.rs:100]: This function can be removed.

**Review by jkilpatr — CHANGES_REQUESTED** [review]
Looks good, just needs a few functions cleaned up.

File: rita_client/src/exit_manager/mod.rs (hunk 2 above)
Comment (ptulugu) [reply rita_client/src/exit_manager/mod.rs]: removed

File: rita_client/src/exit_manager/mod.rs (hunk 3 above)
Comment (ptulugu) [reply rita_client/src/exit_manager/mod.rs]: removed

File: rita_client/src/exit_manager/mod.rs (hunk 4 above)
Comment (ptulugu) [reply rita_client/src/exit_manager/mod.rs]: fixed

File: rita_client/src/exit_manager/mod.rs (hunk 5 above)
Comment (ptulugu) [reply rita_client/src/exit_manager/mod.rs]: removed

**Review by jkilpatr — APPROVED** [review]

**Pull request merged**